src/material/dialog/dialog-content-directives.ts

```diff
@@ -121,6 +121,9 @@ export class MatDialogTitle implements OnInit, OnDestroy {
         // Note: we null check the queue, because there are some internal
         // tests that are mocking out `MatDialogRef` incorrectly.
         this._dialogRef._containerInstance?._ariaLabelledByQueue?.push(this.id);
+        // Updating state requires calling `markForCheck` in order to ensure
+        // state is synchronized to the DOM.
+        this._dialogRef?._containerInstance.changeDetectorRef.markForCheck();
```

Comment:
Rather than exposing the `ChangeDetectorRef`, we should add a method on the dialog container like `_assignAriaLabelledBy`. It would allow us to make the `_ariaLabelledByQueue` private as well.